### PR TITLE
User story 6 authorization

### PR DESCRIPTION
**Pull request by 7022Andre**
- Add authorization for lists and items
- Fix rspec test to work with devise auth

### CODE REVIEW

File: app/controllers/api/items_controller.rb [hunk 1]
```diff
@@ -32,4 +47,11 @@ def update
   def item_params
     params.require(:item).permit(:body, :completed)
   end
+
+  def user_authorized?
+    list = Item.find(params[:id]).list_id
+    user_id = List.find(list).user_id
+
+    current_user.id == user_id
```
Comment (wilsonrector) [inline app/controllers/api/items_controller.rb]:
```
def user_authorized?
  list = Item.find(params[:id]).list
  current_user == list.user || not_authorized
end
```

File: app/models/list.rb [hunk 2]
```diff
@@ -1,6 +1,6 @@
 class List < ApplicationRecord
   belongs_to :user
-  has_many :items, dependent: :destroy
+  has_many :items, dependent: :delete_all
```
Comment (wilsonrector) [inline app/models/list.rb]: Change back

File: app/models/user.rb [hunk 3]
```diff
@@ -1,7 +1,7 @@
-class User < ActiveRecord::Base
+class User < ApplicationRecord
   devise :database_authenticatable, :registerable,
          :trackable, :validatable
   include DeviseTokenAuth::Concerns::User
 
-  has_many :lists, dependent: :destroy
+  has_many :lists, dependent: :delete_all
```
Comment (wilsonrector) [inline app/models/user.rb]: Change back

**Comment by 7022Andre** [comment]
Refactored code and fixed issues with models and test.

**Pull request merged**